sys/include/syslog.h

```diff
@@ -0,0 +1,276 @@
+#endif
+
+/**
+ * @brief   Log a message in the syslog
+ *
+ * @param[in]   entry       log entry to use
+ * @param[in]   priority    priority of the log message
+ * @param[in]   format      message format string
+ */
+void syslog(syslog_entry_t *entry, int priority, const char *format, ...);
+
+/**
+ * @brief   Log a message in the syslog
+ *
+ * @param[in]   entry       log entry to use
+ * @param[in]   priority    priority of the log message
+ * @param[in]   format      message format string
+ * @param[in]   ap          parameter list
+ */
+void vsyslog(syslog_entry_t *entry, int priority, const char *format, va_list ap);
+
+/**
+ * @brief   Set the log level for a log entry
+ *
+ * @param[in]   entry       log entry to use
+ * @param[in]   mask        log level, 0 is ignored
+ *
+ * @return previous mask of the entry
+ */
+int setlogmask(syslog_entry_t *entry, int mask);
+
+/**
+ * @brief   Initialize the syslog module
+ */
+void syslog_init(void);
+
+/**
+ * @brief   Set syslog hostname
+ *
+ * @paran[in]   name    hostname to use, the buffer needs to remain allocated
+ */
+void syslog_set_hostname(const char *name);
+
+/**
+ * @brief   Change the log level of all entries
+ *
+ * @param[in]   level       log level, 0 is ignored
+ *
+ * @return previous greatest level of all entries
+ */
+int syslog_set_level(int level);
+
+/**
+ * @brief   Returns the current greatest log level
+ *
+ * @return log level
+ */
+int syslog_get_level(void);
+
+/**
+ * @brief   Get the next log in the entry list
+ *
+ * @param[in]   log     current log, NULL is passed for the first call
+ *
+ * @return  next entry in the list, NULL if the end of the list is reached
+ */
+const syslog_entry_t *syslog_get_next_log(const syslog_entry_t *log);
+
+/**
+ * @brief   Get the current mask of a given entry
+ *
+ * @param[in]   log     log entry
+ *
+ * @return log level
+ */
+int syslog_getmask(const syslog_entry_t *log);
+
+/**
+ * @brief   Get ident of a given entry
+ *
+ * @param[in]   log     log entry
+ *
+ * @return ident string
+ */
+const char *syslog_getident(const syslog_entry_t *log);
+
+/* Backend-specific functions */
+
+/* STDIO */
+
+/**
+ * @brief   Enable/disable stdio output
+ *
+ * @param[in]   print   true to enable stdio output, false to disable
+ */
+void syslog_backend_stdio_print(bool print);
+
+/* File */
+
+/**
+ * @brief   Start file backend
+ *
+ * Start syslog file backend. This should be called when the filesystem is
+ * ready. It will initialize the file backend by looking into the log directory
+ * and counting the current number of files.
+ */
+int syslog_backend_file_start(void);
+
+/**
+ * @brief   Suspend log rotation
+ *
+ * If called multiple times, syslog_backend_file_resume_rotation() must be
+ * called the same number of times so the rotation is resumed.
+ */
+void syslog_backend_file_suspend_rotation(void);
+
+/**
+ * @brief   Resume log rotation
+ */
+void syslog_backend_file_resume_rotation(void);
```

Comment:
I want to be able to upload files to a backend so that the easiest way of ensuring the file is not removed during the upload operation, while logging is not completely stopped.